packages/codegen/src/EntityDbMetadata.ts

```diff
@@ -216,11 +216,16 @@ function newOneToOne(entity: Entity, r: O2MRelation): OneToOneField {
 }
 
 /** Returns the collection name to use on `entity` when referring to `otherEntity`s. */
-export function collectionName(entity: Entity, otherEntity: Entity): string {
-  // TODO Handle multiple fks from otherEntity --> entity
+export function collectionName(entity: Entity, otherEntity: Entity, r: M2ORelation | O2MRelation): string {
   // TODO Handle conflicts in names
   // I.e. if the other side is `child.project_id`, use `children`.
   let fieldName = otherEntity.name;
+  // check if we multiple FKs from otherEntity --> entity and prefix with FK name if so
+  const sourceTable = r instanceof M2ORelation ? r.sourceTable : r.targetTable;
```

Comment:
I love everything about pg-structure except for their sourceTable/targetTable naming. I swear its backwards. And maybe only backwards some of the time, which is even worse than being backwards all of the time. I'm honestly not sure which is which.